src/atomic-components/headingText.tsx

```diff
@@ -0,0 +1,61 @@
+import * as React from "react";
+import { ComponentConfig, Fields } from "@measured/puck";
+import { Heading, HeadingProps, headingVariants } from "./atoms/heading.js";
+import { useDocument } from "../hooks/useDocument.js";
+import {
+  YextEntityField,
+  YextEntityFieldSelector,
+} from "../components/YextEntityFieldSelector.js";
+import { EntityField } from "../components/EntityField.js";
+import { resolveYextEntityField } from "../utils/resolveYextEntityField.js";
+
+export interface HeadingTextProps extends HeadingProps {
+  text: YextEntityField;
+}
+
+const HeadingText = React.forwardRef<HTMLHeadingElement, HeadingTextProps>(
+  ({ text, ...headingProps }, ref) => {
+    const document = useDocument();
+
+    return (
+      <EntityField fieldId={text.field}>
+        <Heading ref={ref} {...headingProps}>
+          {resolveYextEntityField(document, text)}
+        </Heading>
+      </EntityField>
+    );
+  }
+);
+
+HeadingText.displayName = "HeadingText";
+
+const headingTextFields: Fields<HeadingTextProps> = {
```

Comment:
I think we want the 4 properties listed in the doc under Theme Styling -> Heading Text to be props here too. Allows the user to set the globally or at the component level

Reply:
Oh hmmm I was looking at Components -> Heading Text for the two props settable by users. 

I was thinking Theme Styling -> Heading Text was the theme props?

Might be misunderstanding

Reply:
Yeah sorry that's what I had thought at first but then Aaron & I talked about this some last Friday and I guess it didn't make it into the doc. [Here's](https://yext.slack.com/archives/C06A06BCUUF/p1729524337505309?thread_ts=1729285727.447869&cid=C06A06BCUUF) him talking about it for another component. 

Basically, Theme Manager will set a default styles for each but then the component can override. You already have that for color and size. (Also text transform doesn't work in TM because of tailwind stuff :/ so it has to be on the component level)

Reply:
Ah gotcha